src/main/java/org/spongepowered/api/data/key/Key.java

```diff
@@ -49,6 +50,14 @@
     Class<V> getValueClass();
 
     /**
+     * Gets the class of the element of the {@link Value} this {@link Key}
+     * is representing.
+     *
+     * @return The element class
+     */
+    Class<?> getElementClass();
```

Comment:
I would rather not use type tokens because of their requirement that people use them as asnonymous classes to get the generic parameters or fully understand the limitations of type tokens in the first place. The end result still is the same: in a pure abstract system, the only real use for this method is validation at runtime, not compile time.